### PR TITLE
APEXCORE-533 layering of the properties in case of config apps were …

### PR DESCRIPTION
**Pull request by sandeshh**
…failing, because of the order in which they are processed, now fixed

@davidyan74 please review.

### CODE REVIEW

File: engine/src/test/java/com/datatorrent/stram/cli/ApexCliTest.java [hunk 1]
```diff
@@ -195,6 +195,7 @@ public void testLaunchAppPackagePrecedenceWithConfigPackage() throws Exception
     Assert.assertEquals("app-package-config", props.get("dt.test.3"));
     Assert.assertEquals("user-home-config", props.get("dt.test.4"));
     Assert.assertEquals("package-default", props.get("dt.test.5"));
+    Assert.assertEquals("fromConfigPackage", props.get("dt.test.7"));
```
Comment (davidyan74) [inline engine/src/test/java/com/datatorrent/stram/cli/ApexCliTest.java]: Please try to use the same convention for the property value in this test.

File: engine/src/test/resources/testAppPackage/mydtapp/src/main/resources/META-INF/properties.xml [hunk 2]
```diff
@@ -45,6 +45,10 @@
     <value>package-default</value>
   </property>
   <property>
+    <name>dt.test.7</name>
+    <value>fromAppPackage</value>
```
Comment (davidyan74) [inline engine/src/test/resources/testAppPackage/mydtapp/src/main/resources/META-INF/properties.xml]: There is no test that verifies this value

File: engine/src/test/resources/testConfigPackage/testConfigPackageSrc/META-INF/properties.xml [hunk 3]
```diff
@@ -48,5 +48,9 @@
     <name>dt.test.2</name>
     <value>config-package</value>
   </property>
+  <property>
+    <name>dt.test.7</name>
```
Comment (davidyan74) [inline engine/src/test/resources/testConfigPackage/testConfigPackageSrc/META-INF/properties.xml]: What is the purpose of this property? It seems to serve the same purpose as dt.test.2?

**Review by davidyan74 — CHANGES_REQUESTED** [review]

**Comment by sandeshh** [comment]
@davidyan74 addressed all the review comments.

File: engine/src/main/java/com/datatorrent/stram/client/AppPackage.java [hunk 4]
```diff
@@ -404,6 +407,8 @@ public void processAppDirectory(boolean skipJars)
         LOG.warn("Ignoring file {} with unknown extension in app directory", entry.getName());
       }
     }
+
+    processAppProperties();
```
Comment (davidyan74) [inline engine/src/main/java/com/datatorrent/stram/client/AppPackage.java:411]: Does moving the processAppProperties here mean that the properties specified in the app json file or the app properties files would be overwritten by the properties-\* xml files?

Comment (davidyan74) [reply engine/src/main/java/com/datatorrent/stram/client/AppPackage.java:411]: Ignore the above comment.

File: engine/src/test/java/com/datatorrent/stram/cli/ApexCliTest.java [hunk 5]
```diff
@@ -206,6 +206,22 @@ public void testLaunchAppPackagePrecedenceWithConfigPackage() throws Exception
   }
 
   @Test
+  public void testLaunchAppPackagePrecedenceWithConfigPackageApps() throws Exception
+  {
+    // set launch command options
+    ApexCli.LaunchCommandLineInfo commandLineInfo = ApexCli
+        .getLaunchCommandLineInfo(new String[]{"-D", "dt.test.1=launch-define", "-apconf", "my-app-conf1.xml", "-conf", configFile.getAbsolutePath(), "-useConfigApps", "exclusive"});
+    // process and look at launch config
```
Comment (sandeshh) [inline engine/src/test/java/com/datatorrent/stram/cli/ApexCliTest.java:214]: @davidyan74 I added _-apConf_ and updated the tests.

**Pull request merged**